core/bin/external_node/src/main.rs

```diff
@@ -488,16 +482,36 @@ async fn main() -> anyhow::Result<()> {
     .await
     .context("failed to build a connection_pool")?;
 
+    let main_node_url = config
+        .required
+        .main_node_url()
+        .expect("Main node URL is incorrect");
+    tracing::info!("Main node URL is: {main_node_url}");
+    let main_node_client = <dyn MainNodeClient>::json_rpc(&main_node_url)
+        .context("Failed creating JSON-RPC client for main node")?;
+    let reverter = BlockReverter::new(
+        NodeRole::External,
+        config.required.state_cache_path.clone(),
+        config.required.merkle_tree_path.clone(),
+        None,
+        connection_pool.clone(),
+        L1ExecutedBatchesRevert::Allowed,
+    );
+
+    let mut reorg_detector = ReorgDetector::new(main_node_client.clone(), connection_pool.clone());
+    match reorg_detector.check_consistency().await {
```

Comment:
Probably it makes sense to add a comment here explaining that we're checking for the reorg in the beginning because we expect that if reorg is detected during the node lifecycle, the node will exit the same way as it does with any other critical error, and would restart. Then, on the 2nd launch reorg would be detected here, and processed, and the node will be able to operate normally.

Reply:
done